ddlambda.go

```diff
@@ -44,6 +44,9 @@ type (
 
 		// DebugLogging will turn on extended debug logging.
 		DebugLogging bool
+
+		// Trace library operations with AWS X-Ray.
+		Trace bool
```

Comment:
Can we give this a more descriptive name. Maybe TraceWrapper? Also, for context, down the track we are going to support dd-trace-go in addition to X-Ray in this library. So maybe keep the description here more generic.